src/System/Console/GetOpt/Generics.hs

```diff
@@ -15,8 +15,8 @@
 {-# LANGUAGE TypeSynonymInstances  #-}
 {-# LANGUAGE ViewPatterns          #-}
 
-{-# OPTIONS_GHC -fno-warn-unrecognised-pragmas #-}
 {-# OPTIONS_GHC -fno-warn-deprecated-flags #-}
+{-# OPTIONS_GHC -fno-warn-unrecognised-pragmas #-}
```

Comment:
This just made my day. It saves a lot of CPP when getting things to compile with both 7.8 and 7.10 with overlapping instances.

Though I would go the other way, and CPP OverlappingInstances and this for only >= 7.8, and then also have the `{-# OVERLAPPABLE #-}` pragmas. That way you can still remain aware (and document in your code) what it is that you expect to be overlapping.

None of this is relevant to the code review per se, since not using CPP seems like a big plus.